app/controllers/chargeback_controller.rb

```diff
@@ -713,21 +713,14 @@ def get_tags_all(category)
 
   def get_cis_all
     @edit[:cb_assign][:cis] = {}
-    if @edit[:new][:cbshow_typ] == "enterprise"
-      e_id = MiqEnterprise.first
-      @edit[:cb_assign][:cis]["#{e_id.id}"] = "Enterprise"
-    elsif @edit[:new][:cbshow_typ] == "storage"
-      Storage.all.each do |s|
-        @edit[:cb_assign][:cis][s.id] = s.name
-      end
-    elsif @edit[:new][:cbshow_typ] == "ext_management_system"
-      ExtManagementSystem.all.each do |ms|
-        @edit[:cb_assign][:cis][ms.id] = ms.name
-      end
-    elsif @edit[:new][:cbshow_typ] == "ems_cluster"
-      EmsCluster.all.each do |cl|
-        @edit[:cb_assign][:cis][cl.id] = cl.name
+    classtype =
+      if @edit[:new][:cbshow_typ] == "enterprise"
+        MiqEnterprise
+      else
+        @edit[:new][:cbshow_typ].classify.constantize
```

Comment:
```
app/controllers/chargeback_controller.rb
759:    @edit[:new][:cbshow_typ] = params[:cbshow_typ] if params[:cbshow_typ]
```
this worries me

Yes, we need a whitelist or this is an invitation for some unwanted experimentation by some nosy hacker